File: bahasaKorea_Android/bahasaKorea/app/src/main/java/com/im/daeseong/bahasakorea/BahasaAPI/kataApi.java
package com.im.daeseong.bahasakorea.BahasaAPI;

import com.im.daeseong.bahasakorea.Database.KataItems;

import java.util.HashMap;
import java.util.List;

public class kataApi {

    private static kataApi mApi = null;

    private HashMap<Integer, KataItems> hangulItemHashMap;

    public kataApi(){
        hangulItemHashMap = new HashMap<Integer, KataItems>();
    }

    public static kataApi getInstance() {
        if(mApi == null){
            mApi = new kataApi();
        }
        return mApi;
    }

    public void InitData(List<KataItems> items){

        try {

            //List<KataItems> items = kataManager.getKata();
            for(int i=0; i<items.size(); i++){
                KataItems item = items.get(i);
                int rIndex = item.getrIndex();
                String KataKor = item.getKataKor();
                String KataIndo = item.getKataIndo();
                String KataEng = item.getKataEng();
                hangulItemHashMap.put(i, new KataItems(rIndex, KataKor, KataIndo, KataEng));
            }
            items.clear();

        }catch (Exception e){
        }
    }

    public KataItems get(int nItem){
        return hangulItemHashMap.get(nItem);
    }

    public int size(){
        return hangulItemHashMap.size();
    }

    public void clear(){
        hangulItemHashMap.clear();
    }

}

File: bahasakorea(kotlin)/app/src/main/java/com/daeseong/bahasakorea/BahasaAPI/kataApi.kt
package com.daeseong.bahasakorea.BahasaAPI

import com.daeseong.bahasakorea.Database.KataItems
import java.util.*

class kataApi {
    
    private val hangulItemHashMap: HashMap<Int, KataItems> = HashMap<Int, KataItems>()

    companion object {
        private var instance: kataApi? = null
        fun getInstance(): kataApi {
            if (instance == null) {
                instance = kataApi()
            }
            return instance as kataApi
        }
    }
        
    fun InitData(items: MutableList<KataItems>) {

        try {
            
            for (i in items.indices) {
                val item: KataItems = items[i]
                val rIndex: Int = item.rIndex
                val KataKor: String = item.kataKor
                val KataIndo: String = item.kataIndo
                val KataEng: String = item.kataEng
                hangulItemHashMap[i] = KataItems(rIndex, KataKor, KataIndo, KataEng)
            }
            items.clear()            
        } catch (e: Exception) {
        }
    }

    operator fun get(nItem: Int): KataItems? {
        return hangulItemHashMap[nItem]
    }

    fun size(): Int {
        return hangulItemHashMap.size
    }

    fun clear() {
        hangulItemHashMap.clear()
    }
}
